src/renderers/dom/shared/eventPlugins/ChangeEventPlugin.js

```diff
@@ -351,6 +351,11 @@ var ChangeEventPlugin = {
         targetInst
       );
     }
+
+    // When blurring, set the value attribute for number inputs
+    if (topLevelType === 'topBlur' && targetNode.type === 'number') {
+      targetNode.setAttribute('value', targetNode.value);
+    }
   },
```

Comment:
Basically, onBlur just sets the target node's value attribute to it's current value, but only if it's a number input. I don't know that this is the best place to put this. Any suggestions? 

cc @spicyj ?